### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
This PR provides several improvements:
- Subscribe workspaces that are already starting to the EnvironmentOutput channel. 
- Add  'show-workspaces' and 'restart-workspace' handlers.
- Fixed styles.
- Fixed error message.

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/19070
fixes https://github.com/eclipse/che/issues/19174
fixes https://github.com/eclipse/che/issues/18917
fixes https://github.com/eclipse/che/issues/18589
fixes https://github.com/eclipse/che/issues/19234

Screenshot from 2021-03-05 03-53-45
Screenshot from 2021-03-05 03-54-15
Screenshot from 2021-03-05 17-53-08

Signed-off-by: Oleksii Orel <oorel@redhat.com>

### CODE REVIEW

File: src/containers/IdeLoader.tsx [hunk 1]
```diff
@@ -119,8 +121,41 @@ class IdeLoaderContainer extends React.PureComponent<Props, State> {
     }
   }
 
+  private async handleMessage(event: MessageEvent): Promise<void> {
+    if (!event || typeof event.data !== 'string') {
+      return;
```
Comment (sleshchenko) [inline src/containers/IdeLoader.tsx:126]: When it's going to happen? Would be it a good idea to at least log console warning?

**Review by sleshchenko — COMMENTED** [review]
I found another tiny bug which would be good to fix but not critical, \n are not displayed correctly:

Screenshot_20210305_113520
the message from Che Server
```
{"jsonrpc":"2.0","method":"runtime/log","params":{"time":"2021-03-05T09:37:41.809521613Z","text":"List of plugins and editors to install\n- redhat/java/latest - Java Linting, Intellisense, formatting, refactoring, Maven/Gradle support and more...\n- eclipse/che-machine-exec-plugin/nightly - Che Plug-in with che-machine-exec service to provide creation terminal or tasks for Eclipse Che workspace containers.\n- eclipse/che-theia/next - Eclipse Theia, get the latest release each day.\n","runtimeId":{"ownerId":"a2f45de2-006b-4b46-a3eb-66db9f286f72","envName":"default","workspaceId":"workspace37y0tb6hifcdrdk3"}}}
```

**Comment by sleshchenko** [comment]
Seems this PR introduces a bug with multiple subscription to the logs which leads to duplicates lines, after I restarted workspace more than one in the same tab

Screenshot_20210305_114539
Screenshot_20210305_114550

**Comment by sleshchenko** [comment]
Close workspace works as expected, but I was not able to enable plugins to make Theia restart a workspace
Screenshot_20210305_115021

File: src/containers/IdeLoader.tsx (hunk 1 above)
Comment (olexii4) [reply src/containers/IdeLoader.tsx:126]: The data sent by the message emitter can be any data type.
https://developer.mozilla.org/en-US/docs/Web/API/MessageEvent/data

It is always possible. We do not need any warnings in this case.

But the additional check ```!event ``` is not necessary. I have removed it.

File: src/store/Workspaces/index.ts [hunk 2]
```diff
@@ -408,7 +439,7 @@ const mapMerge = (originMap: Map<string, string[]>, additionalMap: Map<string, s
   const res = new Map<string, string[]>();
   originMap.forEach((val: string[], key: string) => {
     const merge = (val: string[], newVal: string[] | undefined): string[] => {
-      if (!newVal || (val.length > 0 && newVal.length === 1 && val[val.length - 1] === newVal[0])) {
+      if (!newVal || (val.length > 0 && newVal.length === 1)) {
```
Comment (akurinnoy) [inline src/store/Workspaces/index.ts:442]: we definitely need a unit test for this function

File: src/containers/IdeLoader.tsx [hunk 3]
```diff
@@ -119,8 +121,41 @@ class IdeLoaderContainer extends React.PureComponent<Props, State> {
...
+      this.props.history.replace({ pathname });
+      window.postMessage('show-navbar', '*');
+    } else if (event.data.startsWith('restart-workspace:')) {
+      const { allWorkspaces, match: { params } } = this.props;
+      const workspace = allWorkspaces.find(workspace =>
+        workspace.namespace === params.namespace && workspace.devfile.metadata.name === this.workspaceName);
+      if (!workspace) {
+        return;
+      }
+      const res = event.data.split(':');
```
[the remark this answers is not in the document]
Comment (olexii4) [reply src/containers/IdeLoader.tsx]: Ok. I have added ```const [,workspaceId, machineToken] = event.data.split(':');```

File: src/containers/IdeLoader.tsx [hunk 4]
```diff
@@ -119,8 +121,41 @@ class IdeLoaderContainer extends React.PureComponent<Props, State> {
     }
   }
 
+  private async handleMessage(event: MessageEvent): Promise<void> {
+    if (typeof event.data !== 'string') {
+      return;
+    }
+    if (event.data === 'show-workspaces') {
+      const pathname = buildWorkspacesPath();
+      this.props.history.replace({ pathname });
```
[the remark this answers is not in the document]
Comment (olexii4) [reply src/containers/IdeLoader.tsx]: Ok. I have changed it.

File: src/containers/IdeLoader.tsx [hunk 5]
```diff
@@ -119,8 +121,41 @@ class IdeLoaderContainer extends React.PureComponent<Props, State> {
...
+        console.error('Machine token validation failed. ', error);
+      }
+    }
+  }
+
   public async componentWillUnmount(): Promise<void> {
     this.debounce.unsubscribeAll();
+    window.removeEventListener(
+      'message',
+      event => this.handleMessage(event)
```
[the remark this answers is not in the document]
Comment (olexii4) [reply src/containers/IdeLoader.tsx]: Good catch. I have fixed it.

File: src/services/validate-token/index.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import axios from 'axios';
+
+export async function validateMachineToken(workspaceId: string, machineToken?: string): Promise<void> {
```
[the remark this answers is not in the document]
Comment (olexii4) [reply src/services/validate-token/index.ts]: Ok. I have changed it.

**Comment by akurinnoy** [comment]
@olexii4 after the last commit I can't reproduce [this issue](https://github.com/eclipse/che-dashboard/pull/181#issuecomment-795176281).

**Review by akurinnoy — APPROVED** [review]

**Pull request merged**